gui/tooltips.lua

```diff
@@ -0,0 +1,281 @@
+    local job = unit.job
+    if job and job.current_job then
+        return dfhack.job.getName(job.current_job)
+    end
+    return nil
+end
+
+local function GetUnitNameAndJob(unit)
+    local sb = {}
+    sb[#sb+1] = dfhack.units.getReadableName(unit)
+    local jobName = GetUnitJob(unit)
+    if jobName then
+        sb[#sb+1] = ": "
+        sb[#sb+1] = jobName
+    end
+    return table.concat(sb)
+end
+
+local function GetTooltipText(x,y,z)
+    local txt = {}
+    local units = dfhack.units.getUnitsInBox(x,y,z,x,y,z) or {} -- todo: maybe (optionally) use filter parameter here?
+
+    for _,unit in pairs(units) do
+        txt[#txt+1] = GetUnitNameAndJob(unit)
+        txt[#txt+1] = NEWLINE
+    end
+
+    return txt
+end
+
+--------------------------------------------------------------------------------
+-- MouseTooltip is an almost copy&paste of the DimensionsTooltip
+--
+if RELOAD then MouseTooltip = nil end
+MouseTooltip = defclass(MouseTooltip, ResizingPanel)
+
+MouseTooltip.ATTRS{
+    frame_style=gui.FRAME_THIN,
+    frame_background=gui.CLEAR_PEN,
+    no_force_pause_badge=true,
+    auto_width=true,
+    display_offset={x=3, y=3},
+}
+
+function MouseTooltip:init()
+    ensure_key(self, 'frame').w = 17
+    self.frame.h = 4
+
+    self.label = widgets.Label{
+        frame={t=0},
+        auto_width=true,
+    }
+
+    self:addviews{
+        widgets.Panel{
+            -- set minimum size for tooltip frame so the DFHack frame badge fits
+            frame={t=0, l=0, w=7, h=2},
+        },
+        self.label,
+    }
+end
+
+function MouseTooltip:render(dc)
+    if not follow_mouse then return end
+
+    local x, y = dfhack.screen.getMousePos()
+    if not x then return end
+
+    local pos = dfhack.gui.getMousePos()
+    local text = GetTooltipText(pos2xyz(pos))
+    if #text == 0 then return end
+    self.label:setText(text)
+
+    local sw, sh = dfhack.screen.getWindowSize()
+    local frame_width = math.max(9, self.label:getTextWidth() + 2)
+    self.frame.l = math.min(x + self.display_offset.x, sw - frame_width)
+    self.frame.t = math.min(y + self.display_offset.y, sh - self.frame.h)
+    self:updateLayout()
+    MouseTooltip.super.render(self, dc)
+end
+
+--------------------------------------------------------------------------------
+
+if RELOAD then TooltipsVizualizer = nil end
+TooltipsVizualizer = defclass(TooltipsVizualizer, gui.ZScreen)
+TooltipsVizualizer.ATTRS{
+    focus_path='TooltipsVizualizer',
+    pass_movement_keys=true,
+}
+
+function TooltipsVizualizer:init()
+    local controls = TooltipControlWindow{view_id = 'controls'}
+    local tooltip = MouseTooltip{view_id = 'tooltip'}
+    self:addviews{controls, tooltip}
+end
+
+-- map coordinates -> interface layer coordinates
+function GetScreenCoordinates(map_coord)
+    if not map_coord then return end
+
+    -- -> map viewport offset
+    local vp = df.global.world.viewport
+    local vp_Coord = vp.window_x -- is actually coord
```

Comment:
this field appears to be misnamed -- let me correct that to just `window` (or maybe "corner" since that's how it's named in the DF headers)

https://github.com/DFHack/df-structures/pull/818

Reply:
the structure field rename PR has been merged

Reply:
you can subscribe to the `testing` or `testing-adv-beta` branch on Steam to get the latest structures and APIs